src/mem/alloc.rs

```diff
@@ -0,0 +1,1063 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright (c) 2022 SUSE LLC
+ *
+ * Author: Joerg Roedel <jroedel@suse.de>
+ *
+ * vim: ts=4 sw=4 et
+ */
+
+use crate::cpu::vc_early_make_pages_private;
+use crate::dyn_mem_begin;
+use crate::dyn_mem_end;
+use crate::globals::*;
+use crate::mem::{pgtable_pa_to_va, pgtable_va_to_pa};
+use crate::util::locking::SpinLock;
+use core::alloc::{GlobalAlloc, Layout};
+use core::mem::size_of;
+use core::ptr;
+use x86_64::addr::{align_up, PhysAddr, VirtAddr};
+use x86_64::structures::paging::frame::PhysFrame;
+
+struct PageStorageType(u64);
+
+// Support only order 0 allocations for now
+pub const MAX_ORDER: usize = 6;
```

Comment:
If this is a compile-time configurable parameter, we should check inside mem_init() that the assigned value is not bigger than the real limit. This is, STATIC_ASSERT!(MAX_ORDER < REAL_MAX_ORDER) where REAL_MAX_ORDER=6